dashboard/app/controllers/api_controller.rb

```diff
@@ -259,6 +262,96 @@ def section_assessments
     render json: data
   end
 
+  # Return results for surveys, which are long-assessment LevelGroup levels with the anonymous property.
+  # At least five students in the section must have submitted answers.  The answers for each contained
+  # sublevel are shuffled randomly.
+  def section_surveys
```

Comment:
I did an even more dramatic refactoring last night which nests the iteration to correspond to the result set, which makes the code a lot simpler to read.  It also avoids parsing the main LevelGroup's last_attempt, and instead retrieves the per-sublevel last_attempts instead.  (We know they exist because surveys were implemented after we started writing such attempts.)